primitives/account/src/account/staking_contract/validator.rs

```diff
@@ -488,7 +488,7 @@ impl StakingContract {
         }
 
         // Check that the signer is correct.
-        if *signer != Address::from(&validator.signing_key) {
+        if *signer != Address::from(&PublicKey::EdDSA(validator.signing_key)) {
```

Comment:
Maybe we should implement `Address::from` not only for `PublicKey` but also the different pk types. Then we don't need to wrap the key just for getting the address.

Reply:
It's already implemented, but these occurances seem to be leftovers from an early draft where they were not implemented.